spec/features/guide_publishing_flow_spec.rb

```diff
@@ -7,59 +7,79 @@
     allow_any_instance_of(GuidePublisher).to receive(:process)
   end
 
-  it "should create a new edition if there are no drafts" do
-    guide = given_a_published_guide_exists
-    visit guides_path
-    link = there_should_be_a_control_link "Create new edition", document: guide
-    link.click
-    the_form_should_be_prepopulated
-    fill_in "Title", with: "Sample Published Edition 2"
-    click_button "Save Draft"
-    expect(current_path).to eq root_path
+  context "latest edition is published" do
+    it "should create a new draft edition if the latest edition is published" do
+      guide = given_a_published_guide_exists
 
-    expect(guide.editions.published.size).to eq 1
-    expect(guide.editions.draft.size).to eq 1
-    expect(guide.editions.map(&:title)).to match_array ["Sample Published Edition", "Sample Published Edition 2"]
+      publisher_double = double(:publisher)
+      expect(GuidePublisher).to receive(:new).with(guide: guide).and_return(publisher_double)
+      expect(publisher_double).to receive(:process)
+
+      visit guides_path
+      click_button "Create new edition"
+      the_form_should_be_prepopulated
+      expect(guide.editions.published.size).to eq 1
+      expect(guide.editions.draft.size).to eq 1
+    end
   end
 
-  it "should create a new edition even when saving a draft" do
-    guide = given_a_guide_exists state: 'draft'
-    visit guides_path
-    link = there_should_be_a_control_link "Continue editing", document: guide
-    link.click
-    fill_in "Title", with: "Sample Published Edition 2"
-    click_button "Save Draft"
-    expect(current_path).to eq root_path
+  context "latest edition is not published" do
+    it "should not create a new draft edition" do
```

Comment:
"should not save an extra draft if someone else clicks the link in the meantime" ?